### PR TITLE
cmake: enable Fortran before checking for the compiler

### PR DESCRIPTION
**Pull request by junghans**
**Summary**

`chain.x` didn't be build unless some other package incidentally enabled fortran.

**Related Issues**

**Author(s)**

@junghans 

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] The feature has been verified to work with the CMake based build system

**Further Information, Files, and Links**

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -553,11 +553,15 @@ if(BUILD_TOOLS)
   install(TARGETS binary2txt DESTINATION ${CMAKE_INSTALL_BINDIR})
 
   include(CheckGeneratorSupport)
-  if(CMAKE_GENERATOR_SUPPORT_FORTRAN AND CMAKE_Fortran_COMPILER)
+  if(CMAKE_GENERATOR_SUPPORT_FORTRAN)
     enable_language(Fortran)
```
Comment (junghans) [inline cmake/CMakeLists.txt:557]: The only thing I am not sure about is, if this call will be a fatal error if there is no fortran compiler.

Comment (junghans) [reply cmake/CMakeLists.txt:557]: Just checked yes it does...

Comment (akohlmey) [reply cmake/CMakeLists.txt]: so we need to check before enabling:
https://cmake.org/cmake/help/v3.10/module/CheckLanguage.html#module:CheckLanguage

Comment (junghans) [reply cmake/CMakeLists.txt]: Fixed.

**Comment by junghans** [comment]
Ha, I came up with exactly the same idea, but you pushed it earlier.

**Pull request merged**